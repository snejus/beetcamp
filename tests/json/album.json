{
  "@context": "https://schema.org",
  "@id": "https://ute-rec.bandcamp.com/album/ute004",
  "@type": "MusicAlbum",
  "additionalProperty": [
    {
      "@type": "PropertyValue",
      "name": "art_id",
      "value": 1035657740
    },
    {
      "@type": "PropertyValue",
      "name": "featured_track_num",
      "value": 1
    },
    {
      "@type": "PropertyValue",
      "name": "license_name",
      "value": "all_rights_reserved"
    }
  ],
  "albumRelease": [
    {
      "@id": "https://ute-rec.bandcamp.com/album/ute004",
      "@type": [
        "MusicRelease",
        "Product"
      ],
      "additionalProperty": [
        {
          "@type": "PropertyValue",
          "name": "item_id",
          "value": 2990976741
        },
        {
          "@type": "PropertyValue",
          "name": "item_type",
          "value": "a"
        },
        {
          "@type": "PropertyValue",
          "name": "selling_band_id",
          "value": 788052312
        },
        {
          "@type": "PropertyValue",
          "name": "type_name",
          "value": "Digital"
        },
        {
          "@type": "PropertyValue",
          "name": "art_id",
          "value": 1035657740
        }
      ],
      "description": "Includes high-quality download in MP3, FLAC and more. Paying supporters also get unlimited streaming via the free Bandcamp app.",
      "image": [
        "https://f4.bcbits.com/img/a1035657740_10.jpg"
      ],
      "musicReleaseFormat": "DigitalFormat",
      "name": "UTE004"
    },
    {
      "@id": "https://ute-rec.bandcamp.com/album/ute004#p122109000",
      "@type": [
        "MusicRelease",
        "Product"
      ],
      "additionalProperty": [
        {
          "@type": "PropertyValue",
          "name": "item_id",
          "value": 122109000
        },
        {
          "@type": "PropertyValue",
          "name": "item_type",
          "value": "p"
        },
        {
          "@type": "PropertyValue",
          "name": "selling_band_id",
          "value": 788052312
        },
        {
          "@type": "PropertyValue",
          "name": "type_name",
          "value": "Vinyl LP"
        },
        {
          "@type": "PropertyValue",
          "name": "image_ids",
          "value": [
            22500386,
            22500378
          ]
        },
        {
          "@type": "PropertyValue",
          "name": "is_music_merch",
          "value": true
        },
        {
          "@type": "PropertyValue",
          "name": "type_id",
          "value": 2
        }
      ],
      "image": [
        "https://f4.bcbits.com/img/0022500386_10.jpg",
        "https://f4.bcbits.com/img/0022500378_10.jpg"
      ],
      "musicReleaseFormat": "VinylFormat",
      "name": "12\" Vinyl",
      "offers": {
        "@type": "Offer",
        "additionalProperty": [
          {
            "@type": "PropertyValue",
            "name": "includes_digital_download",
            "value": true
          }
        ],
        "availability": "SoldOut",
        "price": 12.0,
        "priceCurrency": "EUR",
        "priceSpecification": {
          "minPrice": 12.0
        },
        "url": "https://ute-rec.bandcamp.com/album/ute004#p122109000-buy"
      }
    },
    {
      "@id": "https://ute-rec.bandcamp.com/track/the-human-experience-empathy-mix",
      "@type": "MusicRelease"
    },
    {
      "@id": "https://ute-rec.bandcamp.com/track/parallell",
      "@type": "MusicRelease"
    },
    {
      "@id": "https://ute-rec.bandcamp.com/track/formulae",
      "@type": "MusicRelease"
    },
    {
      "@id": "https://ute-rec.bandcamp.com/track/biotope",
      "@type": "MusicRelease"
    }
  ],
  "albumReleaseType": "AlbumRelease",
  "byArtist": {
    "@type": "MusicGroup",
    "name": "Mikkel Rev"
  },
  "copyrightNotice": "All Rights Reserved",
  "creditText": "Production: Mikkel Haraldstad\r\nMaster: Joel Krozer, 6bitdeep\r\nGraphics: Haider Ahmed",
  "dateModified": "19 Feb 2023 14:55:18 GMT",
  "datePublished": "17 Jul 2020 00:00:00 GMT",
  "description": "Our dear Mikkel serving four emotional and warm trancers focused on the journey within.",
  "image": "https://f4.bcbits.com/img/a1035657740_10.jpg",
  "keywords": [
    "Oslo"
  ],
  "mainEntityOfPage": "https://ute-rec.bandcamp.com/album/ute004",
  "name": "UTE004",
  "numTracks": 4,
  "publisher": {
    "@id": "https://ute-rec.bandcamp.com",
    "@type": "MusicGroup",
    "additionalProperty": [
      {
        "@type": "PropertyValue",
        "name": "band_id",
        "value": 788052312
      },
      {
        "@type": "PropertyValue",
        "name": "has_any_downloads",
        "value": true
      },
      {
        "@type": "PropertyValue",
        "name": "has_download_codes",
        "value": true
      },
      {
        "@type": "PropertyValue",
        "name": "image_height",
        "value": 2048
      },
      {
        "@type": "PropertyValue",
        "name": "image_id",
        "value": 23711597
      },
      {
        "@type": "PropertyValue",
        "name": "image_width",
        "value": 2048
      }
    ],
    "description": "✣ Record label\n✣ Club nights\n✣ Forest gatherings",
    "foundingLocation": {
      "@type": "Place",
      "name": "Oslo, Norway"
    },
    "image": "https://f4.bcbits.com/img/0023711597_10.jpg",
    "mainEntityOfPage": [
      {
        "@type": "WebPage",
        "name": "SoundCloud",
        "url": "https://soundcloud.com/ute-rec"
      }
    ],
    "name": "Ute.Rec",
    "subjectOf": [
      {
        "@type": "WebPage",
        "additionalProperty": [
          {
            "@type": "PropertyValue",
            "name": "nav_type",
            "value": "a"
          }
        ],
        "name": "Sub-labels",
        "url": "https://ute-rec.bandcamp.com/artists"
      },
      {
        "@type": "WebPage",
        "additionalProperty": [
          {
            "@type": "PropertyValue",
            "name": "nav_type",
            "value": "m"
          }
        ],
        "name": "Releases",
        "url": "https://ute-rec.bandcamp.com/music"
      },
      {
        "@type": "WebPage",
        "additionalProperty": [
          {
            "@type": "PropertyValue",
            "name": "nav_type",
            "value": "p"
          }
        ],
        "name": "Merch",
        "url": "https://ute-rec.bandcamp.com/merch"
      }
    ]
  },
  "track": {
    "@type": "ItemList",
    "itemListElement": [
      {
        "@type": "ListItem",
        "item": {
          "@id": "https://ute-rec.bandcamp.com/track/the-human-experience-empathy-mix",
          "@type": "MusicRecording",
          "additionalProperty": [
            {
              "@type": "PropertyValue",
              "name": "track_id",
              "value": 2780084730
            },
            {
              "@type": "PropertyValue",
              "name": "license_name",
              "value": "all_rights_reserved"
            }
          ],
          "copyrightNotice": "All Rights Reserved",
          "duration": "P00H08M24S",
          "mainEntityOfPage": "https://ute-rec.bandcamp.com/track/the-human-experience-empathy-mix",
          "name": "The Human Experience (Empathy Mix)"
        },
        "position": 1
      },
      {
        "@type": "ListItem",
        "item": {
          "@id": "https://ute-rec.bandcamp.com/track/parallell",
          "@type": "MusicRecording",
          "additionalProperty": [
            {
              "@type": "PropertyValue",
              "name": "track_id",
              "value": 4132926702
            },
            {
              "@type": "PropertyValue",
              "name": "license_name",
              "value": "all_rights_reserved"
            }
          ],
          "copyrightNotice": "All Rights Reserved",
          "duration": "P00H08M07S",
          "mainEntityOfPage": "https://ute-rec.bandcamp.com/track/parallell",
          "name": "Parallell"
        },
        "position": 2
      },
      {
        "@type": "ListItem",
        "item": {
          "@id": "https://ute-rec.bandcamp.com/track/formulae",
          "@type": "MusicRecording",
          "additionalProperty": [
            {
              "@type": "PropertyValue",
              "name": "track_id",
              "value": 2058726302
            },
            {
              "@type": "PropertyValue",
              "name": "license_name",
              "value": "all_rights_reserved"
            }
          ],
          "copyrightNotice": "All Rights Reserved",
          "duration": "P00H07M11S",
          "mainEntityOfPage": "https://ute-rec.bandcamp.com/track/formulae",
          "name": "Formulae"
        },
        "position": 3
      },
      {
        "@type": "ListItem",
        "item": {
          "@id": "https://ute-rec.bandcamp.com/track/biotope",
          "@type": "MusicRecording",
          "additionalProperty": [
            {
              "@type": "PropertyValue",
              "name": "track_id",
              "value": 3564730108
            },
            {
              "@type": "PropertyValue",
              "name": "license_name",
              "value": "all_rights_reserved"
            }
          ],
          "copyrightNotice": "All Rights Reserved",
          "duration": "P00H07M01S",
          "mainEntityOfPage": "https://ute-rec.bandcamp.com/track/biotope",
          "name": "Biotope"
        },
        "position": 4
      }
    ],
    "numberOfItems": 4
  }
}
