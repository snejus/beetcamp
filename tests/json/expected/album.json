[
  {
    "album": "UTE004",
    "album_id": "https://ute-rec.bandcamp.com/album/ute004",
    "albumdisambig": null,
    "albumstatus": "Official",
    "albumtype": "album",
    "albumtypes": [
      "album"
    ],
    "artist": "Mikkel Rev",
    "artist_credit": null,
    "artist_id": "https://ute-rec.bandcamp.com",
    "artist_sort": null,
    "artists": [
      "Mikkel Rev"
    ],
    "artists_credit": [],
    "artists_ids": [
      "https://ute-rec.bandcamp.com"
    ],
    "artists_sort": [],
    "asin": null,
    "barcode": null,
    "catalognum": "UTE004",
    "comments": "Our dear Mikkel serving four emotional and warm trancers focused on the journey within.\n---\nProduction: Mikkel Haraldstad\nMaster: Joel Krozer, 6bitdeep\nGraphics: Haider Ahmed",
    "country": "NO",
    "cover_art_url": "https://f4.bcbits.com/img/a1035657740_10.jpg",
    "data_source": "bandcamp",
    "data_url": "https://ute-rec.bandcamp.com/album/ute004",
    "day": 17,
    "discogs_albumid": null,
    "discogs_artistid": null,
    "discogs_labelid": null,
    "genre": null,
    "label": "Ute.Rec",
    "language": null,
    "media": "Digital Media",
    "mediums": 1,
    "month": 7,
    "original_day": 17,
    "original_month": 7,
    "original_year": 2020,
    "release_group_title": null,
    "releasegroup_id": null,
    "releasegroupdisambig": null,
    "script": null,
    "style": null,
    "tracks": [
      {
        "album": "UTE004",
        "arranger": null,
        "artist": "Mikkel Rev",
        "artist_credit": null,
        "artist_id": "https://ute-rec.bandcamp.com",
        "artist_sort": null,
        "artists": [
          "Mikkel Rev"
        ],
        "artists_credit": [],
        "artists_ids": [
          "https://ute-rec.bandcamp.com"
        ],
        "artists_sort": [],
        "bpm": null,
        "composer": null,
        "composer_sort": null,
        "data_source": "bandcamp",
        "data_url": "https://ute-rec.bandcamp.com/album/ute004",
        "disctitle": null,
        "genre": null,
        "index": 1,
        "initial_key": null,
        "length": 504,
        "lyricist": null,
        "mb_workid": null,
        "media": "Digital Media",
        "medium": 1,
        "medium_index": 1,
        "medium_total": 4,
        "release_track_id": null,
        "title": "The Human Experience (Empathy Mix)",
        "track_alt": null,
        "track_id": "https://ute-rec.bandcamp.com/track/the-human-experience-empathy-mix",
        "work": null,
        "work_disambig": null
      },
      {
        "album": "UTE004",
        "arranger": null,
        "artist": "Mikkel Rev",
        "artist_credit": null,
        "artist_id": "https://ute-rec.bandcamp.com",
        "artist_sort": null,
        "artists": [
          "Mikkel Rev"
        ],
        "artists_credit": [],
        "artists_ids": [
          "https://ute-rec.bandcamp.com"
        ],
        "artists_sort": [],
        "bpm": null,
        "composer": null,
        "composer_sort": null,
        "data_source": "bandcamp",
        "data_url": "https://ute-rec.bandcamp.com/album/ute004",
        "disctitle": null,
        "genre": null,
        "index": 2,
        "initial_key": null,
        "length": 487,
        "lyricist": null,
        "mb_workid": null,
        "media": "Digital Media",
        "medium": 1,
        "medium_index": 2,
        "medium_total": 4,
        "release_track_id": null,
        "title": "Parallell",
        "track_alt": null,
        "track_id": "https://ute-rec.bandcamp.com/track/parallell",
        "work": null,
        "work_disambig": null
      },
      {
        "album": "UTE004",
        "arranger": null,
        "artist": "Mikkel Rev",
        "artist_credit": null,
        "artist_id": "https://ute-rec.bandcamp.com",
        "artist_sort": null,
        "artists": [
          "Mikkel Rev"
        ],
        "artists_credit": [],
        "artists_ids": [
          "https://ute-rec.bandcamp.com"
        ],
        "artists_sort": [],
        "bpm": null,
        "composer": null,
        "composer_sort": null,
        "data_source": "bandcamp",
        "data_url": "https://ute-rec.bandcamp.com/album/ute004",
        "disctitle": null,
        "genre": null,
        "index": 3,
        "initial_key": null,
        "length": 431,
        "lyricist": null,
        "mb_workid": null,
        "media": "Digital Media",
        "medium": 1,
        "medium_index": 3,
        "medium_total": 4,
        "release_track_id": null,
        "title": "Formulae",
        "track_alt": null,
        "track_id": "https://ute-rec.bandcamp.com/track/formulae",
        "work": null,
        "work_disambig": null
      },
      {
        "album": "UTE004",
        "arranger": null,
        "artist": "Mikkel Rev",
        "artist_credit": null,
        "artist_id": "https://ute-rec.bandcamp.com",
        "artist_sort": null,
        "artists": [
          "Mikkel Rev"
        ],
        "artists_credit": [],
        "artists_ids": [
          "https://ute-rec.bandcamp.com"
        ],
        "artists_sort": [],
        "bpm": null,
        "composer": null,
        "composer_sort": null,
        "data_source": "bandcamp",
        "data_url": "https://ute-rec.bandcamp.com/album/ute004",
        "disctitle": null,
        "genre": null,
        "index": 4,
        "initial_key": null,
        "length": 421,
        "lyricist": null,
        "mb_workid": null,
        "media": "Digital Media",
        "medium": 1,
        "medium_index": 4,
        "medium_total": 4,
        "release_track_id": null,
        "title": "Biotope",
        "track_alt": null,
        "track_id": "https://ute-rec.bandcamp.com/track/biotope",
        "work": null,
        "work_disambig": null
      }
    ],
    "va": false,
    "year": 2020
  },
  {
    "album": "UTE004",
    "album_id": "https://ute-rec.bandcamp.com/album/ute004#p122109000",
    "albumdisambig": null,
    "albumstatus": "Official",
    "albumtype": "album",
    "albumtypes": [
      "album"
    ],
    "artist": "Mikkel Rev",
    "artist_credit": null,
    "artist_id": "https://ute-rec.bandcamp.com",
    "artist_sort": null,
    "artists": [
      "Mikkel Rev"
    ],
    "artists_credit": [],
    "artists_ids": [
      "https://ute-rec.bandcamp.com"
    ],
    "artists_sort": [],
    "asin": null,
    "barcode": null,
    "catalognum": "UTE004",
    "comments": "Our dear Mikkel serving four emotional and warm trancers focused on the journey within.\n---\nProduction: Mikkel Haraldstad\nMaster: Joel Krozer, 6bitdeep\nGraphics: Haider Ahmed",
    "country": "NO",
    "cover_art_url": "https://f4.bcbits.com/img/a1035657740_10.jpg",
    "data_source": "bandcamp",
    "data_url": "https://ute-rec.bandcamp.com/album/ute004",
    "day": 17,
    "discogs_albumid": null,
    "discogs_artistid": null,
    "discogs_labelid": null,
    "genre": null,
    "label": "Ute.Rec",
    "language": null,
    "media": "Vinyl",
    "mediums": 1,
    "month": 7,
    "original_day": 17,
    "original_month": 7,
    "original_year": 2020,
    "release_group_title": null,
    "releasegroup_id": null,
    "releasegroupdisambig": null,
    "script": null,
    "style": null,
    "tracks": [
      {
        "album": "UTE004",
        "arranger": null,
        "artist": "Mikkel Rev",
        "artist_credit": null,
        "artist_id": "https://ute-rec.bandcamp.com",
        "artist_sort": null,
        "artists": [
          "Mikkel Rev"
        ],
        "artists_credit": [],
        "artists_ids": [
          "https://ute-rec.bandcamp.com"
        ],
        "artists_sort": [],
        "bpm": null,
        "composer": null,
        "composer_sort": null,
        "data_source": "bandcamp",
        "data_url": "https://ute-rec.bandcamp.com/album/ute004",
        "disctitle": "12\" Vinyl",
        "genre": null,
        "index": 1,
        "initial_key": null,
        "length": 504,
        "lyricist": null,
        "mb_workid": null,
        "media": "Vinyl",
        "medium": 1,
        "medium_index": 1,
        "medium_total": 4,
        "release_track_id": null,
        "title": "The Human Experience (Empathy Mix)",
        "track_alt": null,
        "track_id": "https://ute-rec.bandcamp.com/track/the-human-experience-empathy-mix",
        "work": null,
        "work_disambig": null
      },
      {
        "album": "UTE004",
        "arranger": null,
        "artist": "Mikkel Rev",
        "artist_credit": null,
        "artist_id": "https://ute-rec.bandcamp.com",
        "artist_sort": null,
        "artists": [
          "Mikkel Rev"
        ],
        "artists_credit": [],
        "artists_ids": [
          "https://ute-rec.bandcamp.com"
        ],
        "artists_sort": [],
        "bpm": null,
        "composer": null,
        "composer_sort": null,
        "data_source": "bandcamp",
        "data_url": "https://ute-rec.bandcamp.com/album/ute004",
        "disctitle": "12\" Vinyl",
        "genre": null,
        "index": 2,
        "initial_key": null,
        "length": 487,
        "lyricist": null,
        "mb_workid": null,
        "media": "Vinyl",
        "medium": 1,
        "medium_index": 2,
        "medium_total": 4,
        "release_track_id": null,
        "title": "Parallell",
        "track_alt": null,
        "track_id": "https://ute-rec.bandcamp.com/track/parallell",
        "work": null,
        "work_disambig": null
      },
      {
        "album": "UTE004",
        "arranger": null,
        "artist": "Mikkel Rev",
        "artist_credit": null,
        "artist_id": "https://ute-rec.bandcamp.com",
        "artist_sort": null,
        "artists": [
          "Mikkel Rev"
        ],
        "artists_credit": [],
        "artists_ids": [
          "https://ute-rec.bandcamp.com"
        ],
        "artists_sort": [],
        "bpm": null,
        "composer": null,
        "composer_sort": null,
        "data_source": "bandcamp",
        "data_url": "https://ute-rec.bandcamp.com/album/ute004",
        "disctitle": "12\" Vinyl",
        "genre": null,
        "index": 3,
        "initial_key": null,
        "length": 431,
        "lyricist": null,
        "mb_workid": null,
        "media": "Vinyl",
        "medium": 1,
        "medium_index": 3,
        "medium_total": 4,
        "release_track_id": null,
        "title": "Formulae",
        "track_alt": null,
        "track_id": "https://ute-rec.bandcamp.com/track/formulae",
        "work": null,
        "work_disambig": null
      },
      {
        "album": "UTE004",
        "arranger": null,
        "artist": "Mikkel Rev",
        "artist_credit": null,
        "artist_id": "https://ute-rec.bandcamp.com",
        "artist_sort": null,
        "artists": [
          "Mikkel Rev"
        ],
        "artists_credit": [],
        "artists_ids": [
          "https://ute-rec.bandcamp.com"
        ],
        "artists_sort": [],
        "bpm": null,
        "composer": null,
        "composer_sort": null,
        "data_source": "bandcamp",
        "data_url": "https://ute-rec.bandcamp.com/album/ute004",
        "disctitle": "12\" Vinyl",
        "genre": null,
        "index": 4,
        "initial_key": null,
        "length": 421,
        "lyricist": null,
        "mb_workid": null,
        "media": "Vinyl",
        "medium": 1,
        "medium_index": 4,
        "medium_total": 4,
        "release_track_id": null,
        "title": "Biotope",
        "track_alt": null,
        "track_id": "https://ute-rec.bandcamp.com/track/biotope",
        "work": null,
        "work_disambig": null
      }
    ],
    "va": false,
    "year": 2020
  }
]
